appinventor/components/src/com/google/appinventor/components/runtime/ListAdapterWithRecyclerView.java

```diff
@@ -0,0 +1,347 @@
+        layoutParams2.topMargin = 10;
+        textViewSecond.setLayoutParams(layoutParams2);
+        textViewSecond.setTextSize(textSize);       
+        textViewSecond.setTextColor(textColor);
+                
+        ImageView imageView = new ImageView(context);
+        idImages=View.generateViewId();
+        imageView.setId(idImages);
+        LinearLayout.LayoutParams layoutParamsImage = new LinearLayout.LayoutParams(imageWidth, imageHeight);
+        imageView.setLayoutParams(layoutParamsImage);
+
+        LinearLayout linearLayout2= new LinearLayout(context);
+        LinearLayout.LayoutParams layoutParamslinear2 = new LinearLayout.LayoutParams(0,LinearLayout.LayoutParams.WRAP_CONTENT,2);
+        linearLayout2.setLayoutParams(layoutParamslinear2);
+        linearLayout2.setOrientation(LinearLayout.VERTICAL);
+
+        LinearLayout linearLayout1= new LinearLayout(context);
+        LinearLayout.LayoutParams layoutParamslinear1 = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT,LinearLayout.LayoutParams.WRAP_CONTENT);
+        linearLayout1.setLayoutParams(layoutParamslinear1);
+        linearLayout1.setOrientation(LinearLayout.HORIZONTAL);
+
+        linearLayout2.addView(textViewFirst);
+        linearLayout2.addView(textViewSecond);
+        linearLayout1.addView(imageView);
+        linearLayout1.addView(linearLayout2);
+        cardView.setLayoutParams(params1);
+        cardView.addView(linearLayout1);
+        }
+
+        return new RvViewHolder(cardView);
+    }
+
+    @Override
+    public void onBindViewHolder(final RvViewHolder holder, int position) {
+         
+            final int pos=position;
+
+            holder.cardView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                    holder.onClick(v);
+                    selection[pos]=!selection[pos];                   
+                    if(selection[pos]){
+                    holder.cardView.setBackgroundColor(selectionColor);
+                    }else{
+                    holder.cardView.setBackgroundColor(backgroundColor);
+                    }
+                }
+            });
+            if(layoutType==Component.LISTVIEW_LAYOUT_SINGLE_TEXT){
+            String first =firstItem[position];
+            holder.textViewFirst.setText(first);
+            }else if(layoutType==Component.LISTVIEW_LAYOUT_TWO_TEXT){
+            String first =firstItem[position];
+            String second=secondItem[position];
+            
+            holder.textViewFirst.setText(first);
+            holder.textViewSecond.setText(second);
+            }else if(layoutType==Component.LISTVIEW_LAYOUT_TWO_TEXT_LINEAR){
+            String first =firstItem[position];
+            String second=secondItem[position];
+            String combined = first + second;
```

Comment:
There should be a space inserted between first and second strings.

A different option that occurred to me is that you could keep both textViewFirst and textViewSecond, but turn off content wrapping in the LayoutParams object when you define those LinearLayout objects in the RvViewHolder. This would, I think, cut the content off at the end of one line rather than wrapping the text strangely. You could then  use both the Text and DetailText color/size parameters.

I can see arguments for either solution.